// Copyright (C) 2019-2022 The Xaya developers
// Distributed under the MIT software license, see the accompanying
// file COPYING or http://www.opensource.org/licenses/mit-license.php.

syntax = "proto2";

package xaya.proto;

/** A piece of data with signatures of channel participants.  */
message SignedData
{

  /**
   * The data itself.  This can encode different entities, like board states,
   * moves or changes made to a channel in agreement.
   */
  optional bytes data = 1;

  /**
   * The signatures made on the data by channel participants.  Each signature
   * stores raw bytes based on the underlying message signing/verification
   * provider used.  For instance, for signing based on Xaya Core's RPC
   * interface with signmessage/verifymessage, the data is the base64-decoded
   * signature string as raw binary.
   *
   * For the message, the channel's ID (uint256) is concatenated with
   * the base64-encoded reinit ID, a nul byte, a string describing what the
   * data is (e.g. "state"), another nul byte and finally the value of "data".
   *
   * All this is hashed with SHA-256 and then converted to a lower-case
   * hex string, which is the "message" for "signmessage".
   */
  repeated bytes signatures = 2;

  /**
   * This field is not used in production.  It is a field that should never be
   * set.  We use it to test that function, and that it is called correctly as
   * we need it.
   */
  optional string for_testing_version = 1000;

}
